File: study/inflearn_js/8.재귀함수_완전탐색_DFS/중복순열_구하기.py
n,m  = map(int, input().split())

num = list(range(1,n+1))
ans = [0] * m

def DFS(l):
  global ans
  if l == m:
    print(ans)
    return
  for i in range(n):
    ans[l] = num[i]
    DFS(l+1)
DFS(0)

File: study/inflearn_js/8.재귀함수_완전탐색_DFS/중복순열_구하기.js
/**
 * arr에서 m개를 뽑아서 중복순열을 만들어라.
 */
const arr = [1, 2, 3, 4];
const m = 3;
let answer = "";

function DFS(l, ans) {
  if (l === m) {
    cnt++;
    answer += ans + "\n";
    return;
  }
  for (let i = 0; i < arr.length; i++) {
    DFS(l + 1, !ans ? arr[i] : `${ans} ${arr[i]}`);
  }
}
DFS(0, "");
console.log(answer.trim());
